crates/bevy_core_pipeline/src/smaa/smaa.wgsl

```diff
@@ -0,0 +1,1122 @@
+// Corner Detection Functions
+
+fn detect_horizontal_corner_pattern(weights: vec2<f32>, tex_coord: vec4<f32>, d: vec2<f32>)
+        -> vec2<f32> {
+#ifndef SMAA_DISABLE_CORNER_DETECTION
+    let left_right = step(d.xy, d.yx);
+    var rounding = (1.0 - SMAA_CORNER_ROUNDING_NORM) * left_right;
+
+    rounding /= left_right.x + left_right.y; // Reduce blending for pixels in the center of a line.
+
+    var factor = vec2(1.0, 1.0);
+    factor.x -= rounding.x * sample_edges_level_zero_offset(tex_coord.xy, vec2(0,  1)).r;
+    factor.x -= rounding.y * sample_edges_level_zero_offset(tex_coord.zw, vec2(1,  1)).r;
+    factor.y -= rounding.x * sample_edges_level_zero_offset(tex_coord.xy, vec2(0, -2)).r;
+    factor.y -= rounding.y * sample_edges_level_zero_offset(tex_coord.zw, vec2(1, -2)).r;
+
+    return weights * saturate(factor);
+#else   // SMAA_DISABLE_CORNER_DETECTION
+    return weights;
+#endif  // SMAA_DISABLE_CORNER_DETECTION
+}
+
+fn detect_vertical_corner_pattern(weights: vec2<f32>, tex_coord: vec4<f32>, d: vec2<f32>)
+        -> vec2<f32> {
+#ifndef SMAA_DISABLE_CORNER_DETECTION
+    let left_right = step(d.xy, d.yx);
+    var rounding = (1.0 - SMAA_CORNER_ROUNDING_NORM) * left_right;
+
+    rounding /= left_right.x + left_right.y;
+
+    var factor = vec2(1.0, 1.0);
+    factor.x -= rounding.x * sample_edges_level_zero_offset(tex_coord.xy, vec2( 1, 0)).g;
+    factor.x -= rounding.y * sample_edges_level_zero_offset(tex_coord.zw, vec2( 1, 1)).g;
+    factor.y -= rounding.x * sample_edges_level_zero_offset(tex_coord.xy, vec2(-2, 0)).g;
+    factor.y -= rounding.y * sample_edges_level_zero_offset(tex_coord.zw, vec2(-2, 1)).g;
+
+    return weights * saturate(factor);
+#else   // SMAA_DISABLE_CORNER_DETECTION
+    return weights;
+#endif  // SMAA_DISABLE_CORNER_DETECTION
+}
+
+//-----------------------------------------------------------------------------
+// Blending Weight Calculation Pixel Shader (Second Pass)
+
+@fragment
+fn blending_weight_calculation_fragment_main(in: BlendingWeightCalculationVaryings)
+        -> @location(0) vec4<f32> {
+    let subsample_indices = vec4(0.0);  // Just pass zero for SMAA 1x, see @SUBSAMPLE_INDICES.
+
+    var weights = vec4(0.0);
+
+    var e = textureSample(edges_texture, edges_sampler, in.tex_coord).rg;
+
+    if (e.g > 0.0) {    // Edge at north
+#ifndef SMAA_DISABLE_DIAG_DETECTION
+        // Diagonals have both north and west edges, so searching for them in
+        // one of the boundaries is enough.
+        weights = vec4(calculate_diag_weights(in.tex_coord, e, subsample_indices), weights.ba);
+
+        // We give priority to diagonals, so if we find a diagonal we skip 
+        // horizontal/vertical processing.
+        let diagonal_found = weights.r + weights.g != 0.0;
+#else   // SMAA_DISABLE_DIAG_DETECTION
+        let diagonal_found = false;
+#endif  // SMAA_DISABLE_DIAG_DETECTION
+
+        if (!diagonal_found) {
```

Comment:
Imo doing an early return if a diagonal is found would be cleaner than this. It'd also let us get rid of the else statement on this function.

Reply:
OK, I changed it to early return.